flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/triggers/ProcessingTimeoutTrigger.java

```diff
@@ -0,0 +1,100 @@
+package org.apache.flink.streaming.api.windowing.triggers;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.state.ValueState;
+import org.apache.flink.api.common.state.ValueStateDescriptor;
+import org.apache.flink.api.common.typeutils.base.LongSerializer;
+import org.apache.flink.streaming.api.windowing.windows.Window;
+
+/**
+ * A {@link Trigger} that can turn any {@link Trigger} into a timeout {@code Trigger}.
+ *
+ * <p>Each record arriving will emit a ProcessingTimeTimer withing the interval,
+ * you can control if the timer will be registered periodic for each event arriving,
+ * by the continualInterval flag.
```

Comment:
Is maybe `resetTimerOnEvent` a better name for `continualInterval`, because that's what it does right? When `continualInterval` is `true` the timer period will be extended on each received event, right?

Reply:
you right about the purpose of the flag. On second thought `continualInterval ` is not a good name, but i changed it to `resetTimerOnNewRecord` , what do you think?